tests/integration/test_charm.py

```diff
@@ -231,9 +230,35 @@ async def test_seldon_alert_rules(ops_test: OpsTest):
     assert seldon_reconcile_error_alert["state"] == "firing"
 
     # cleanup SeldonDeployment
-    client.delete(seldon_deployment, name="seldon-model-1", namespace=namespace)
+    client.delete(seldon_deployment, name="seldon-model-1", namespace=namespace, grace_period=0)
+
+    # wait for application to settle
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60, idle_period=30
+    )
```

Comment:
It takes a bit of time to delete SeldonDeployment. Charm won't see any action, but I want to ensure deployment is deleted, before proceeding to next step.
Deployment shutdown is handled by workload container and that is not in our control.

Reply:
oh ok I see what you're after.

In that case, use `tenacity.retry` instead.  As it is now, `wait_for_idle` is effectively `sleep(30)` and we hope that's enough time.  Use `tenacity.retry` instead.  Retry a function that asserts the resource should not exist here instead of `wait_for_idle`.  Nice thing too is if the deletion hangs, we'll get a clear error message saying so

Reply:
Actually, sleep is fine. It is not more about SeldonDeployment to be remove, but more about Seldon Core workload container to settle which we have no way of checking.

Reply:
Added explicit sleep.

Reply:
Removing it. sleep != wait_for_idle